### PR TITLE
New wof model format

### PR DESCRIPTION
**Pull request by orangejulius**
This is a massive reworking to support the new schema built on WOF data.

Most notably the new schema includes dedicated fields for source and layer, allowing us to avoid a lot of nasty work to reconcile our previous mapping of source and layers to a single field.

Here's an incomplete list of things to be done

- [x] Update API responses to use source and layer from source and layer fields, instead of calculating
- [x] When using source and layer API parameters, use the new fields to filter, instead of only `_type`
- [x] Use new source and layer field to generate GIDs in all endpoints
- [x] update ciao tests
- [x] ensure reverse still works (thank you ciao tests!)
- [x] Create a reasonable set of layer aliases (I think there's only one: coarse)
- [x] Detect and warn when a layer + source combination wouldn't return any results (ex: address layer + quattroshapes)

Connected to #413

### CODE REVIEW

**Comment by dianashk** [comment]
List of things that are broken at the moment, and we should validate have been fixed before we call this done:

- [ ] [autocomplete: 755 5th ave](http://pelias.github.io/compare/#/v1/autocomplete%3Ffocus.point.lat=40.756749&focus.point.lon=-73.998284&text=755%205th%20ave)

**Comment by orangejulius** [comment]
Note to everyone: this code was in a very WIP state when other people started looking at things, so I've done a bunch of cleanup. Enjoy the clean code but you will have to rebase.

File: helper/geojsonify.js [hunk 1]
```diff
@@ -34,24 +33,6 @@ function lookupSource(src) {
  * Elasticsearch document source allows a more specific layer name to be chosen
  */
 function lookupLayer(src) {
-  if (src._type === 'geoname') {
```
Comment (dianashk) [inline helper/geojsonify.js]: yay for deleting complicated old code!

File: controller/place.js [hunk 2]
```diff
@@ -13,40 +13,16 @@ function setup( backend ){
...
       return {
         _index: 'pelias',
-        _type: id.type,
+        _type: id.layers,
         _id: id.id
       };
     });
 
     service.mget( backend, query, function( err, docs ) {
+      console.log('err:' + err);
```
Comment (orangejulius) [inline controller/place.js]: this should go away

File: service/mget.js [hunk 3]
```diff
@@ -24,6 +24,7 @@ function service( backend, query, cb ){
 
   // query new backend
   backend().client.mget( cmd, function( err, data ){
+    console.log('service err: ' + err);
```
Comment (orangejulius) [inline service/mget.js]: get rid of this

File: test/unit/controller/place.js [hunk 4]
```diff
@@ -70,10 +70,13 @@ module.exports.tests.functional_success = function(test, common) {
 module.exports.tests.functional_failure = function(test, common) {
   test('functional failure', function(t) {
     var backend = mockBackend( 'client/mget/fail/1', function( cmd ){
-      t.deepEqual(cmd, { body: { docs: [ { _id: 123, _index: 'pelias', _type: 'b' } ] } }, 'correct backend command');
+      var e = new Error('wtf is going on');
+      console.log(e.stack);
+      console.log('cmd: ' + JSON.stringify(cmd, null, 2));
```
Comment (orangejulius) [inline test/unit/controller/place.js]: this can go away too

File: query/search.js [hunk 5]
```diff
@@ -119,7 +121,9 @@ function generateQuery( clean ){
     textParser( clean.parsed_text, vs );
   }
 
-  return query.render( vs );
+  var q = query.render( vs );
+  //console.log(JSON.stringify(q, null, 2));
+  return q;
```
Comment (orangejulius) [inline query/search.js]: remove

File: sanitiser/_sources_and_layers.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+var _ = require( 'lodash' );
+var type_mapping = require( '../helper/type_mapping' );
+function sanitize( raw, clean ){
+  var messages = { errors: [], warnings: [] };
+
+  var possible_errors = [];
+  var at_least_one_valid_combination = false;
+
+  if (clean.layers && clean.sources) {
```
Comment (orangejulius) [inline sanitiser/_sources_and_layers.js]: peter says add tests

File: sanitiser/_targets.js [hunk 7]
```diff
@@ -1,25 +1,27 @@
...
       targetMap: targetMap,
-      targetMapKeysString: Object.keys(targetMap).join(',')
+      targetMapKeysString: getValidKeys(targetMap)
     });
   };
 }
 
 function sanitize( raw, clean, opts ) {
+  // TODO: remove this next line
+  var propertyName = opts.paramName;//opts.paramName.slice(0,-1);
```
Comment (orangejulius) [inline sanitiser/_targets.js]: remove this

File: controller/search.js [hunk 8]
```diff
@@ -42,7 +47,7 @@ function setup( backend, query ){
         res.data = docs;
         res.meta = meta;
       }
-
+      logger.debug('[ES response]', JSON.stringify(docs));
```
Comment (orangejulius) [inline controller/search.js]: These came in accidentally through a bad rebase. They will go away before merging

File: sanitiser/_sources_and_layers.js (hunk 6 above)
Comment (orangejulius) [inline sanitiser/_sources_and_layers.js]: Tests have now been added

**Comment by dianashk** [comment]
:+1:

**Pull request merged**